streaming/src/event_server.h

```diff
@@ -0,0 +1,227 @@
+#ifndef RAY_STREAMING_EVENT_SERVER_H
+#define RAY_STREAMING_EVENT_SERVER_H
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <unordered_map>
+
+#include "channel.h"
+#include "ring_buffer.h"
+#include "util/streaming_util.h"
+
+namespace ray {
+namespace streaming {
+
+/// Data writer utilizes what's called an event-driven programming model instead
+/// loop forward polling driven that's employed in first simple version.
+/// Event driven model includes two important components: event server and event
+/// queue. In the process of data transmission, the inputer will first define
```

Comment:
What does `inputer` mean?

Reply:
> What does `inputer` mean?
Empty trigger or writing message to ringbuffer.